Add thread run state string to report

DariaKunoichi opened the pull request:
Added a new field to a report with string representation of thread's state.
Adjusted integration tests to check new field.

naftaly commented on Sources/KSCrashRecordingCore/KSThread.c:
```diff
@@ -52,6 +68,34 @@ bool ksthread_getThreadName(const KSThread thread, char *const buffer, int bufLe
     return pthread_getname_np(pthread, buffer, (unsigned)bufLength) == 0;
 }
 
+int ksthread_getThreadState(const KSThread thread)
```
You could simplify this a bunch to the following:

```
static int ksthread_getThreadState(const KSThread thread)
{
    thread_basic_info_t info = {0};
    mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
    kern_return_t kr = 0;
    
    kr = thread_info((thread_t)thread, THREAD_BASIC_INFO, (thread_info_t)info, &count);
    return  kr == KERN_SUCCESS ? info->run_state : TH_STATE_UNSET;
}
```

GLinnik21 replied:
> memory safe function end up calling into vm_read_overwrite

That’s a good point. What solutions are there?

naftaly replied:
I’m not sure. What’s the validation for in these cases anyway? 

I don’t think we need to validate memory returned by these system functions. I imagine those men validation functions were only meant to be used in the memory monitor.

naftaly reviewed (commented):
nice! Thanks for proposing this.

GLinnik21 reviewed (changes requested):
Great thanks!

However, we need to update the minor version of `KSCRASH_REPORT_VERSION` in `Sources/KSCrashRecording/KSCrashReportVersion.h`

DariaKunoichi commented:
Simplified the function a bit according to the review.
Updated minor version in `KSCRASH_REPORT_VERSION`.
Fixed integration tests to skip this feature for watchOS where thread api is unavailable.
Fixed the formatting.

GLinnik21 commented:
> If you meant integration tests then can you point me to where exactly can I set the value if it's not present?

@DariaKunoichi Sorry for the confusion! I messed up. You're right — the `#if !os(watchOS)` conditional is only in the integration tests, not the main library.

> Should I just make it optional in the PartialCrashReport class?

It seems that the crash report format would be better if it could be made consistent for each report version, regardless of the platform. Using platform-specific macros in the decoder might incorrectly assume different report schemas on different platforms, when in reality, the field may or may not be present in any crash report based on runtime conditions.

GLinnik21 commented on Sources/KSCrashRecordingCore/KSThread.c:
```diff
@@ -52,6 +68,30 @@ bool ksthread_getThreadName(const KSThread thread, char *const buffer, int bufLe
+int ksthread_getThreadState(const KSThread thread)
+{
+    integer_t infoBuffer[THREAD_BASIC_INFO_COUNT] = { 0 };
+    thread_basic_info_t info = infoBuffer;
```
I believe pod spec lint problem is here:
` incompatible pointer types initializing 'thread_basic_info_t' with an expression of type 'integer_t[10]’`
